c/serializer/src/agenttypesystem.c

```diff
@@ -1417,10 +1458,40 @@ AGENT_DATA_TYPES_RESULT AgentDataTypes_ToString(STRING_HANDLE destination, const
                         {
                             result = AGENT_DATA_TYPES_OK;
                         }
+
+                        // Clean up temp buffer if allocated
+                        if (tempBuffer != NULL) {
+                            free(tempBuffer);
+                            tempBuffer = NULL;
```

Comment:
I think you can skip the "else if"in this case and have  {} for the else statement. And then a one liner free on the else branch of the allocation in line 1433 would do it. Then we don't have a second check here whether tempBuffer has been allocated.